core/runtime/src/main/java/io/quarkiverse/jberet/runtime/QuarkusJobExecutor.java

```diff
@@ -5,14 +5,28 @@
 import org.jberet.spi.JobExecutor;
 import org.wildfly.common.cpu.ProcessorInfo;
 
+import io.quarkus.logging.Log;
 import io.quarkus.runtime.ThreadPoolConfig;
 
 class QuarkusJobExecutor extends JobExecutor {
     private final int maxPoolSize;
 
-    public QuarkusJobExecutor(Executor delegate, final ThreadPoolConfig threadPoolConfig) {
+    public QuarkusJobExecutor(Executor delegate, final ThreadPoolConfig threadPoolConfig, final JBeretConfig config) {
         super(delegate);
-        this.maxPoolSize = threadPoolConfig.maxThreads.orElse(Math.max(8 * ProcessorInfo.availableProcessors(), 200));
+
+        this.maxPoolSize = config.maxAsync().map(max -> {
+            try {
+                var maxAsync = Integer.parseInt(max);
+                if (maxAsync < 1) {
+                    Log.error("max-async value must be 1 or greater if set.");
+                    return null;
+                }
+                return maxAsync + 1; // Adapt for the fact that JBeret requires one thread for coordination.
+            } catch (NumberFormatException ex) {
+                Log.error("max-async configuration value must be numeric! Could not parse value: " + max + " to an integer.");
```

Comment:
static import

maybe param max could be passed as an argument to avoid potential performant issue with string concatenation here :

<img width="699" alt="image" src="https://user-images.githubusercontent.com/8830888/227710781-12edc9d4-aa6c-4428-bc9b-e69f021d2a36.png">